adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -936,22 +936,40 @@ public final void uploadCommandResponseMessage(
                     .withTag(TracingHelper.TAG_AUTHENTICATED.getKey(), authenticatedDevice != null)
                     .start();
 
-            // send response message to application via sender link
-            sendCommandResponse(tenant, commandResponse, currentSpan.context())
-            .map(delivery -> {
-                metrics.incrementCommandResponseDeliveredToApplication(tenant);
-                LOG.trace("delivered command response [command-request-id: {}] to application", commandRequestId);
-                currentSpan.log("delivered command response to application");
-                ctx.response().setStatusCode(HttpURLConnection.HTTP_ACCEPTED);
-                ctx.response().end();
-                return delivery;
+            final Future<TenantObject> tenantConfigTracker = getTenantConfiguration(tenant, currentSpan.context());
+            tenantConfigTracker.compose(tenantObj -> {
+                if (tenantObj.isAdapterEnabled(getTypeName())) {
+                    // send response message to application via sender link
+                    sendCommandResponse(tenant, commandResponse, currentSpan.context())
+                            .map(delivery -> {
+                                metrics.incrementCommandResponseDeliveredToApplication(tenant);
+                                LOG.trace("delivered command response [command-request-id: {}] to application",
+                                        commandRequestId);
+                                currentSpan.log("delivered command response to application");
+                                ctx.response().setStatusCode(HttpURLConnection.HTTP_ACCEPTED);
+                                ctx.response().end();
+                                return delivery;
+                            }).otherwise(t -> {
+                                LOG.debug("could not send command response [command-request-id: {}] to application",
+                                        commandRequestId, t);
+                                TracingHelper.logError(currentSpan, t);
+                                ctx.fail(new ServerErrorException(HttpURLConnection.HTTP_UNAVAILABLE, t));
+                                return null;
+                            }).setHandler(c -> {
+                                currentSpan.finish();
+                            });
+
+                    return Future.succeededFuture();
+                } else {
+                    // this adapter is not enabled for the tenant
+                    return Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_FORBIDDEN,
+                            "adapter is not enabled for tenant"));
+                }
```

Comment:
I would rather have this whole block moved to a subsequent `compose` method invocation, i.e. if the tenant enabled check succeeds, simply return a succeeded future and then do the sending in the next composed block ...